Create Recipe.md (#35)
carpentries/instructor-training-bonus-modules: _extras/Recipe.md

---
layout: page
title: "Online Workshop Recipe"
---

# A Recipe for an Online Carpentries Workshop

> In the past year+ of teaching online, our community has learned many things. Our initial [online training recommendations](https://carpentries.org/online-workshop-recommendations/#planning-your-workshop) recognised that we were in uncharted territory for Carpentries workshops! In consulting with community members, we heard many different narratives and suggestions. Rather than creating a single 'recipe,' we chose to distill that advice into 'things to think about', giving Instructors space to engineer their own solutions to the challenges presented by teaching online.

Reviewer: Think maybe we should change the wording here since it specifically references the last year or so.  Maybe instead to something that references that in 2020 we made a big shift to teaching online and learned a lot.

> In the past year+ of teaching online, our community has learned many things. Our initial [online training recommendations](https://carpentries.org/online-workshop-recommendations/#planning-your-workshop) recognised that we were in uncharted territory for Carpentries workshops! In consulting with community members, we heard many different narratives and suggestions. Rather than creating a single 'recipe,' we chose to distill that advice into 'things to think about', giving Instructors space to engineer their own solutions to the challenges presented by teaching online.

Reviewer 2: I think we should specifically say up front that these are recommendations for teaching technical workshops (or workshops on data and computing skills), since some related ideas (e.g., workflow dependencies) are described later, and because we don't tend to have some features (like helpers) in instructor training.

    * **Meeting 2 (closer to the date)**: Review workshop mechanics, backup planning, learner communications, assessment, and feedback review. Agree on content to cut when time runs short. Test video conferencing features.
    
For more detailed preparation goals, see our [checklist for online workshop preparation](https://carpentries.org/online-workshop-recommendations/#planning-your-workshop).
### Back-channel communications
* **Create a channel** or private conversation thread on The Carpentries Slack workspace (or other preferred tool). 
* **Use your communication channel** to communicate about preparations. This will allow everyone on your instructional team to become familiar with the tool as well as improve your preparations.
### Documentation
* **Create an Etherpad** or other collaborative document, and add all vital information to be included in your workshop introduction for learners who arrive late. This should include:
    * Workshop schedule (including the times of all breaks)
    * Connection instructions
    * What to do in the event of a connection failure
    * How learners can ask questions (e.g. Zoom chat, raise hand, etc.)

>     * Other instructions related to workshop mechanics, including recommended window arrangement

Reviewer 3: maybe also "and optionally logging in from another device to be able to see the screen from learners' perspective"

* **Have a backup** for all roles and documents. Technical failures come in all shapes and sizes, and life can still happen, too!
* **Review your pre-workshop surveys** using the links provided in your email from our workshop administrators. Knowing your audience will help you adjust appropriately to their needs!
### Update your Software
* Especially for Zoom, you will want to be using the most recent version and the same version as everyone else. Remind your learners to update as well!
## 2. Workshop Mechanics
* **Assign a "Zoom DJ"** to assume the "host" role in the meeting. A Zoom DJ will manage breakouts, waiting rooms, muting, and troubleshooting as needed. This role is typically played by a Supporting Instructor; if/when they choose to teach, plan to hand off this role to another Instructor.
* **Customize profile names** to clearly indicate instructors and helpers. For example, a helper named Ada might change their profile to read "Helper: Ada (she/her)"
* **Use Zoom features as you would use sticky notes**, and remember to call upon learners to use them regularly. 

>     * On Zoom, learners can choose "reactions" that include a green '✓' and a red 'x' icon. Unlike other reactions, these persist until cleared.

Reviewer 3: Just to add that most of this document applies to Microsoft Teams, especially after they introduced breakout rooms. It's a bit slower to open/close them then in Zoom, but they work. What I miss in teams is that it has no persistent reactions. A way to simulate them is to paste different messages (e.g. with 'Yes' and 'No' or with answer numbers) in the meeting chat, and ask learners to like the message corresponding to their answer.

* **Use breakout rooms** for socialization and activities whenever possible. These do take time and that may mean cutting content. The benefits to creating a supportive learning environment make this cost worthwhile.
    * Using a shared timer can help, especially as learners get used to using it.

> * **Ask learners to keep the chat focused.** You may wish to provide them with an alternate venue to share social chat.

Reviewer 4: When I taught last year, it was complicated to manage the use of chat. On the one hand, the "social" comments helped the group to consolidate and create a good atmosphere, but on the other hand, it was sometimes distracting. We opted to announce each time it was really necessary to pay attention to the chat (e.g. during an exercise ) and that each person chose to read or not between those moments.

---
layout: page
title: "Online Workshop Recipe"
---

# A Recipe for an Online Carpentries Workshop

> In the past year+ of teaching online, our community has learned many things. Our initial [online training recommendations](https://carpentries.org/online-workshop-recommendations/#planning-your-workshop) recognized that we were in uncharted territory for Carpentries workshops! In consulting with community members, we heard many different narratives and suggestions. Rather than creating a single 'recipe,' we chose to distill that advice into 'things to think about', giving Instructors space to engineer their own solutions to the challenges presented by teaching online.

Reviewer 5: Just a note that we replace z in words like recognized with s - recognised

Author: ```suggestion
In the past year+ of teaching online, our community has learned many things. Our initial [online training recommendations](https://carpentries.org/online-workshop-recommendations/#planning-your-workshop) recognised that we were in uncharted territory for Carpentries workshops! In consulting with community members, we heard many different narratives and suggestions. Rather than creating a single 'recipe,' we chose to distill that advice into 'things to think about', giving Instructors space to engineer their own solutions to the challenges presented by teaching online.
```

> We continue to be excited to learn from the innovations of our Instructor community! However, at this point we have also developed a better idea about the specific advantages and disadvantages of different practices. For those who do not wish to 'reinvent the wheel' but prefer to start with a recommended model, we offer an example structure for an online workshop that we expect to function well.

Reviewer 6: Consider replacing 'reinvent the wheel' as idioms may not translate globally.

> We continue to be excited to learn from the innovations of our Instructor community! However, at this point we have also developed a better idea about the specific advantages and disadvantages of different practices. For those who do not wish to 'reinvent the wheel' but prefer to start with a recommended model, we offer an example structure for an online workshop that we expect to function well.

Reviewer 5: We can embed this link https://docs.carpentries.org/topic_folders/hosts_instructors/index.html#resources-for-online-workshops where the CDT shared online resource from the community

> Note that these recommendations are based characteristics of Zoom, Etherpad, and Slack platforms and may require minor modifications to function on other platforms.

Reviewer 6: ```suggestion
Note that these recommendations are based on the use of Zoom, Etherpad, and Slack platforms and may require minor modifications to function on other platforms.
```

## 1. Advance preparation
### Workshop roles
* **Instructors** (at least 2): take the lead on preparing the curriculum and leading instruction. 

> * **Supporting Instructors** (at least 1): take the lead on preparing the workshop mechanics, coordinating with helpers, and ensuring that everything is working smoothly. They may take on limited instruction and/or an instructional backup role if they choose. They play the role of "Zoom DJ" (below) whenever they are not teaching.

Reviewer 6: A supporting instructor can count as one of your two instructors, right? This makes it seem like you need all three. 

Also (at least one) implies you have to have a supporting instructor. I don't think that is the case, is it?

Author: No -- because the Supporting Instructor role is teaching-optional, they really can't substitute for a second Instructor.

Reviewer 7: Per @sheraaronhurt the supporting instructor role will not continue

### Schedule

> * A **4 half-day** schedule is ideal where practical. This is  helpful in combatting 'zoom fatigue.'

Reviewer 6: Maybe link to a sample 4 day schedule if available

Reviewer 5: Just a side note, 4 days would not work in the South African context, we have found that 5 days are long but leaves the instructors with more room to cover the curriculum and do troubleshooting as well

Author: Is this more inclusive?
```suggestion
* A **4 half-day** (or other time-distributed) schedule is ideal where practical. This is  helpful in combatting 'zoom fatigue.'
```

Reviewer 8: Sounds good :)

* **Schedule plenty of breaks** and stick to them. Remember that breaks can be useful for troubleshooting, too.
* See below with regard to timing challenges. If you are committed to covering the same content that you would in an in-person workshop, you may wish to extend your planned time.
### Planning meetings

> * Plan at **least 2 meetings** before the workshop. 

Reviewer 6: ```suggestion
* Plan at **least 2 meetings** with the Instructional Team (Instructors, Supporting Instructors and Helpers) before the workshop. 
```

Reviewer 5: I have found that a 10min break for every 50min taught was quite effective.

    * **Meeting 1 (well in advance)**: Establish workshop roles, who teaches what, delegate documentation & communications setup (below), plan asynchronous preparation.

>     * **Meeting 2 (closer to the date)**: Review workshop mechanics, backup planning, learner communications, assessment, and feedback review. Agree on content to cut when time runs short. Test video conferencing features.

Reviewer 6: Can we be clearer on what backup planning means?

    
For more detailed preparation goals, see our [checklist for online workshop preparation](https://carpentries.org/online-workshop-recommendations/#planning-your-workshop).
### Back-channel communications
* **Create a channel** or private conversation thread on The Carpentries Slack workspace (or other preferred tool). 
* **Use your communication channel** to communicate about preparations. This will allow everyone on your instructional team to become familiar with the tool as well as improve your preparations.
### Documentation
* **Create an Etherpad** or other collaborative document, and add all vital information to be included in your workshop introduction for learners who arrive late. This should include:

>     * Workshop schedule (including all breaks)

Reviewer 6: ```suggestion
    * Workshop schedule (including the times of all breaks)
```

    * Connection instructions
    * What to do in the event of a connection failure

>     * How to ask questions

Reviewer 6: ```suggestion
    * How learners can ask questions (e.g. Zoom chat, raise hand, etc.)
```

    * Links to The Carpentries Code of Conduct as well as your workshop website
* Add a link to your Etherpad to your **workshop website**.
* For a self-organised workshop, be sure to submit your [workshop request form](https://amy.carpentries.org/forms/self-organised/).
* **Email your learners** with everything they need to know in advance of the workshop. 
    * Our [template for in-person trainings](https://docs.carpentries.org/topic_folders/workshop_administration/email_templates.html#email-learners-before-workshop) may be useful. Much of this info may already be on your Etherpad, so you can be selective about information that needs duplicating.
### Preparing for Adjustments
* **Plan to cut content.** Online workshops often run more slowly than they do in-person; sometimes much more slowly. Decisions to cut back should be intentional and collaborative.
    * Remember that skipping steps in a workflow can create problems with dependencies later on.
* **Agree on your priorities.** Discussing key objectives and broader goals with your instructional team in advance will help everyone navigate difficult trade-offs when time becomes an issue during a workshop.

> * **Have a backup** for all roles and documents. Technical failures come in all shapes and sizes, and life can still happen, too!

Reviewer 5: The scaffolding might also help as a back up to installation issues -https://carpentries.org/blog/2020/04/scaffolds/

* **Review your pre-workshop surveys** using the links provided in your email from our workshop administrators. Knowing your audience will help you adjust appropriately to their needs!
### Update your Software
* Especially for Zoom, you will want to be using the most recent version and the same version as everyone else. Remind your learners to update as well!
## 2. Workshop Mechanics
* **Assign a "Zoom DJ"** to assume the "host" role in the meeting. A Zoom DJ will manage breakouts, waiting rooms, muting, and troubleshooting as needed. This role is typically played by a Supporting Instructor; if/when they choose to teach, plan to hand off this role to another Instructor.

> * **Customize profile names** to clearly indicate instructors and helpers. For example, a helper named Ada might change their profile to read "Helper: Ada (she/her)"

Reviewer 5: Within the African context, we seldom use pronouns. Perhaps it might be a good idea to link why pronouns are used?

* **Use Zoom features as you would use sticky notes**, and remember to call upon learners to use them regularly. 
    * On Zoom, learners can choose "reactions" that include a green '✓' and a red 'x' icon. Unlike other reactions, these persist until cleared.
* **Use breakout rooms** for socialization and activities whenever possible. These do take time and that may mean cutting content. The benefits to creating a supportive learning environment make this cost worthwhile.
    * Using a shared timer can help, especially as learners get used to using it.
* **Ask learners to keep the chat focused.** You may wish to provide them with an alternate venue to share social chat.
* **Use a 3-step help response system**.
    * Step 1: Learners message a helper, who helps via **direct messaging** in the conference chat if they can. 
    * Step 2: Helpers communicate about questions they have received via **back-channel conversations** (even if resolved). This will elevate common questions that merit attention of the teaching Instructor and will allow others to advise if the problem has not been solved.

>     * Step 3: For difficult problems, invite the learner to **share their screen** for live support. Depending on the workshop size, frequency of the problem, etc., this may be done in the main room as a pause in the workshop or in a breakout room with a helper.

Reviewer 5: The instructors can also use the remote control function in Zoom (if activated) to support learners

Reviewer: I wouldn't highly recommend this as it often requires learners to change their system settings to give zoom remote access.  It also feels a bit like taking over a learner's keyboard to me which can be demotivating.

        * If you create breakout rooms for activities, it helps to create extra rooms for this purpose at that time so that activity rooms can be preserved.
        * Keep in mind that learners in breakout rooms will miss ongoing instruction in the main room, so helpers should also help learners to catch up before returning.
* **Divert distractions and disruptions to breaks** wherever possible. Advanced questions and other difficult conversations can be resolved using breakout rooms during breaks. If you have plenty of breaks available, missing one to problem solve will be less strenuous.

## 3. Assessment

> * **Formative assessment is more important than ever**. When teaching online, you lose the ability to visually "read the room," especially if learners' cameras are turned off. That's ok, because frequent formative assessment will provide the feedback you need. 

Reviewer 6: Consider replacing 'read the room' as idioms don't always translate globally

Reviewer 5: Break out rooms caused issues in the South African context. We asked students to share their screens rather and we could all trouble shoot together.

    * Ask a question that learners can answer in the chat.
    * Prepare a polling tool or Google form with a few questions in advance.
    * Use virtual sticky notes to check pacing - 'green check' for 'fine/too slow' and 'red x' for too fast.

>     * Use 'virtual minute cards' for +/- feedback or to ask learners to share 'one thing you have learned'/'one question you still have' at break points

Reviewer 6: I believe there is a virtual minute card template. Maybe  we can link that here

Reviewer 5: Using the nonverbal prompts in Zoom is also helpful for checking in on learners

Reviewer 4: The green check and red x are permanent, so I usually have to remind people to click again after an exercise is finished or the problem is solved.

Reviewer: Link to virtual minute card template - https://docs.google.com/forms/d/1p7iOV5HNvy4POS4g6eottY8RSfKq4kaoKz1-jIFYTMI/template/preview
